packages/cactus-plugin-satp-hermes/src/main/typescript/services/validation/config-validating-functions/validateContractFields.ts

```diff
@@ -0,0 +1,405 @@
+                break;
+              case key.toLowerCase().includes("signature"):
+                if (contractHash === "") {
+                  contractHash = parseAndCheckContractHashV2(
+                    JSON.stringify(jsonData),
+                    "",
+                    false,
+                  );
+                }
+                console.log("Checking the SIGNATURE -", key);
+                checkSignatureV2(contractHash, value, [
+                  loadKeyFromFile("public_key.pem"),
+                ]);
+                break;
+              case key.toLowerCase().includes("contract"):
+                contractLanguage = value;
+                break;
+              case key.toLowerCase().includes("id"):
+                console.log("ContractID", value);
+                break;
+              default:
+                console.log("Unknown field", key);
+                console.log("Content", value);
+            }
+          });
+          return true;
+        }
+        return false;
+      } catch (checkExecErr) {
+        throw checkExecErr;
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+}
+
+
+function checkOntologyV1(ontologyField: string, functionSignatures: string[]) {
+  const ontologyJson = JSON.parse(ontologyField);
+  const ontologyElements = [
+    "lock",
+    "unlock",
+    "mint",
+    "burn",
+    "assign",
+    "approve",
+    "checkpermission",
+  ];
+  Object.entries(ontologyJson).forEach(([key, value]) => {
+    console.log("Checking Ontology Field", key);
+    const keySimplified = key.replace(/[\s+\-_]/g, "").toLowerCase();
+    if (ontologyElements.includes(keySimplified)) {
+      const index = ontologyElements.indexOf(keySimplified);
+      if (index !== -1) {
+        ontologyElements.splice(index, 1);
+      }
+    }
+    if (Array.isArray(value)) {
+      value.forEach((functionElement: any) => {
+        const functionSignature = functionElement.function_signature;
+        const signatureVars = extractElementsFromBrackets(functionSignature);
+        const variables = functionElement.variables;
+        if (
+          signatureVars.length !== 0 &&
+          signatureVars.length !== variables.length
+        ) {
+          throw new Error(
+            "Variables on function signature not coinciding with variables field",
+          );
+        } else {
+          checkOntologyElementV1(keySimplified, variables);
+          functionSignatures.push(functionSignature);
+        }
+      });
+    } else {
+      throw new OntologyFormatError("Function formats not in an array", THIS_FILE_NAME);
+    }
+  });
+  if (ontologyElements.length !== 0) {
+    console.log("Elements missing for:");
+    ontologyElements.forEach((element) => {
+      console.log("->", element);
+    });
+    throw new OntologyFormatError("Not every basic SATP operation is included", THIS_FILE_NAME);
+  }
+}
+
+function checkOntologyElementV1(fieldName: string, variables: string[]) {
```

Comment:
what is difference between v1 and v2? can you extract common points of functionality and re use it (eg adapter pattern)